screens/party/components/meters.lua

```diff
@@ -127,7 +205,10 @@ end
 
 function M.on_input(action_id, action)
 	for _, b in pairs(active_buttons) do
-		gooey.button(b.node, action_id, action, b.func, b.refresh)
+		local button = gooey.button(b.node, action_id, action, b.func, b.refresh)
+		if b.long_pressed_time then
+			gooey_buttons.set_long_pressed_time(button, 0.5)
+		end
```

Comment:
I thought it was weird too, but I don't understand where else it could go.
The only place I can find where you can actually set this property is after `gooey.button()` has been called. You need to set the `long_pressed_time` property on the return value of `gooey.button()` (which is itself a return calue of `core.instance()` inside `gooey/internal/button.lua`). The only place we are calling `gooey.button()` is here, and its API requires you pass `action_id` and `action`, which only exist in `on_input`. So unless there is some other location I am not aware of where we can set the property and have it be picked up by gooey properly, this seems to be the "right" place to do it. Even though, yes, it is strange.

Another curiosity: there is an "internal" button.lua file that has `function BUTTON.set_long_pressed_time(button, time)` in it. But this is not exposed anywhere in the public API for gooey - I actually hacked it myself by adding `M.set_long_pressed_time` to `gooey_buttons` which is part of the app, not part of gooey. This seemed to be following in a pattern of general utility stuff that happened to know about gooey's internal API.

So, in short, if you can guide me to where this property is actually supposed to be set, I'd happily do that. But as it is, the public API of gooey is seems very unusual to me.

Reply:
The `long_pressed_time` work directly on the button that `gooey.button` returns, it's probably a very confusing implementation because the person who actually implemented it is always confused https://github.com/britzl/gooey/pull/46 :joy:

The "normal" way of setting up settings and such is to call the button with the id and it will store that for later, so we can do it in the `ini (or create in this case) and it will still have stored that setting for when calling it in `on_input`

```
M.create(nodes, pokemon_id)
	-- setup all the stuff
	for _, b in pairs(active_buttons) do
		gooey.button(b.node).set_long_pressed_time(0.5)
	end
end
```

Reply:
Oh, okay, I think I see - you can just call `gooey.button` even when not in `on_input`, you just don't pass any actions.